### PR TITLE
Ignoring unknown keys in RPC responses by default

### LINKED ISSUE #1305: Allow ignoring unknown keys in RPC responses
- Remove hardcoded `unknown=EXCLUDE` from deserializing in `FullNodeClient` methods
- Allow constructor argument like `exclude_unknown_fields` in `FullNodeClient`
- Pass the above argument when deserializing RPC responses, make sure that it is propagated to the nested schemas e.g. `TransactionReceipt` and field `execution_resources`

Similar change has been done in https://github.com/software-mansion/starknet-jvm/pull/405

### PR DESCRIPTION
**Pull request by tkumor3**
Closes #1305

- Remove hardcoded unknown=EXCLUDE from deserializing in FullNodeClient
- Ignoring unknown keys in RPC responses by default
- In our test check if the schemas are correct

I had to leave some `unknown=EXCLUDE` because devnet sometimes returns more fields than starknet-spec claim should be sent

## Introduced changes

- 
- 

##

- [ ] This PR contains breaking changes

### CODE REVIEW

**Comment by tkumor3** [comment]
Rest of `unknown="EXCLUDE"` I will remove after merging RPC 0.7.0 development

File: starknet_py/net/full_node_client.py
Comment (ddoktorski) [inline starknet_py/net/full_node_client.py:176]: Why `unknown=EXCLUDE` is not removed from all methods e.g. `get_block_with_txs`?

**Review by ddoktorski — COMMENTED** [review]
Could we add some unit tests by setting a temporary environment variable?

File: starknet_py/net/full_node_client.py
Comment (tkumor3) [reply starknet_py/net/full_node_client.py:176]: The prime problem of this is working with devnet because in a few cases, they are inconsistent with starknet-spec.

**Comment by tkumor3** [comment]
> Could we add some unit tests by setting a temporary environment variable?

It's problematic because we need to keep broken schema in our code to test it

File: starknet_py/net/client_models.py
Comment (THenry14) [inline starknet_py/net/client_models.py:954]: @DelevoXDG please let's make sure we have this field in jvm/swift as well, as per latest discussion on swm-sw channel this is a spec bug

File: starknet_py/net/schemas/rpc.py
Comment (THenry14) [inline starknet_py/net/schemas/rpc.py:967]: we can probably remove this comment/issue, as Ariel already admitted this to be a bug in spec that will be fixed soon

File: starknet_py/__init__.py
Comment (THenry14) [inline starknet_py/__init__.py:3]: just to be double sure - you have checked that this won't overwrite a user setting in script, right? I mean:
```
# import marshmallow
# set Meta to whatever
# import starknet.py
# the rest of the script
```

so the importing starknet.py (and thus this file as well), does not overwrite the user setting

**Review by THenry14 — APPROVED** [review]
lgtm, can be merged after all comments are addressed

File: starknet_py/net/schemas/rpc.py
Comment (tkumor3) [reply starknet_py/net/schemas/rpc.py:967]: I am not sure how they will change L1HandlerTransactionTraceSchema. execution_resources can be required or optional. IMO is better to wait for space change

File: starknet_py/net/full_node_client.py
Comment (THenry14) [reply starknet_py/net/full_node_client.py:177]: you mean starknet-devnet-rs :+1: ok then, the issue of pending+latest blocks was raised a couple of times with the devnet team, so I think we should just wait

File: starknet_py/utils/schema.py
Comment (THenry14) [inline starknet_py/utils/schema.py:6]:
```suggestion
```

File: starknet_py/utils/schema.py
Comment (THenry14) [inline starknet_py/utils/schema.py:11]:
```suggestion
        unknown = RAISE if os.environ.get("STARKNET_PY_MARSHMALLOW_META_MODE") else EXCLUDE
```
or something similar. env variables should always be capital letters, and it may make sense to make it more descriptive + starknet.py specific

**Review by THenry14 — COMMENTED** [review]
To sum up - the now default behaviour will be to configure everything with an env variable, because it is virtually impossible to do so in-script, right?

This PR should be updated with apropriate docs section that will explain this feature and give some examples.

**Comment by tkumor3** [comment]
> This PR should be updated with apropriate docs section that will explain this feature and give some examples.
Any idea about an example, IMO the description how to set it seems to be clear

File: starknet_py/utils/schema.py
Comment (THenry14) [inline starknet_py/utils/schema.py:11]: > This can be achieved by setting the environment variable, STARKNET_PY_MARSHMALLOW_UKNOWN_EXCLUDE,
to true. Consequently, any unknown fields in response will be excluded.

does this code checks this? I'm not sure, but I think it may just check if the env variable is set (so eg `STARKNET_PY_MARSHMALLOW_UKNOWN_EXCLUDE=false` may also set it, resulting in turning on the exclusion). let's double check.

**Comment by THenry14** [comment]
> > This PR should be updated with apropriate docs section that will explain this feature and give some examples.
> 
> Any idea about an example, IMO the description how to set it seems to be clear

The description you've provided should be enough

File: starknet_py/utils/schema.py
Comment (tkumor3) [reply starknet_py/utils/schema.py:14]: This change doesn't work `AttributeError: 'NoneType' object has no attribute 'lower'`

Comment (THenry14) [reply starknet_py/utils/schema.py:14]:
```
str(MARSHMALLOW_UKNOWN_EXCLUDE).lower() == "true"
```
or 
```
(MARSHMALLOW_UKNOWN_EXCLUDE or "").lower() == "true"
```

Comment (THenry14) [reply starknet_py/utils/schema.py:14]: nitpick though

**Review by THenry14 — APPROVED** [review]

**Pull request merged**
